pyuvdata/tests/test_miriad.py

```diff
@@ -19,34 +19,163 @@ def test_ReadMiriadWriteUVFits():
+    uv_in.read_miriad(miriad_file)
+    uv_in.select(times=uv_in.time_array[0])
+    uv_in.select(freq_chans=[0])
+
+    lat, lon, alt = uv_in.telescope_location_lat_lon_alt
+    lat_wrong = lat + 10 * np.pi / 180.
+    uv_in.telescope_location_lat_lon_alt = (lat_wrong, lon, alt)
+    uv_in.write_miriad(latfile)
+    uv_out.read_miriad(latfile)
+
+    lon_wrong = lon + 10 * np.pi / 180.
+    uv_in.telescope_location_lat_lon_alt = (lat, lon_wrong, alt)
+    uv_in.write_miriad(lonfile)
+    uv_out.read_miriad(lonfile)
+
+    uv_in.telescope_location_lat_lon_alt = (lat, lon, alt)
+    uv_in.telescope_name = 'foo'
+    uv_in.write_miriad(telescopefile)
+    uv_out.read_miriad(telescopefile, run_check=False)
+    """
+    uv_in = UVData()
+    latfile = os.path.join(DATA_PATH, 'zen.2456865.60537_wronglat.xy.uvcRREAA')
+    lonfile = os.path.join(DATA_PATH, 'zen.2456865.60537_wronglon.xy.uvcRREAA')
+    telescopefile = os.path.join(DATA_PATH, 'zen.2456865.60537_wrongtelecope.xy.uvcRREAA')
+
+    uvtest.checkWarnings(uv_in.read_miriad, [latfile],
+                         message=['Altitude is not present in file and latitude value does not match'])
+    uvtest.checkWarnings(uv_in.read_miriad, [lonfile],
+                         message=['Altitude is not present in file and longitude value does not match'])
+    uvtest.checkWarnings(uv_in.read_miriad, [telescopefile], {'run_check': False},
+                         nwarnings=2, category=[UserWarning, UserWarning],
+                         message=['Altitude is not present in Miriad file, and telescope',
+                                  'Telescope foo is not in known_telescopes.'])
+
+
+def test_singletimeselect_drift():
+    """
+    Check behavior with writing & reading after selecting a single time from a drift file.
+
+    """
+    uv_in = UVData()
+    uv_out = UVData()
+    miriad_file = os.path.join(DATA_PATH, 'zen.2456865.60537.xy.uvcRREAA')
+    testfile = os.path.join(DATA_PATH, 'test/outtest_miriad.uv')
+    uvtest.checkWarnings(uv_in.read_miriad, [miriad_file],
+                         known_warning='miriad')
+
+    uv_in.select(times=uv_in.time_array[0])
+    uv_in.write_miriad(testfile, clobber=True)
+    uv_out.read_miriad(testfile)
+    nt.assert_equal(uv_in, uv_out)
+
+    # check that setting the phase_type works
+    uv_out.read_miriad(testfile, phase_type='drift')
+    nt.assert_equal(uv_in, uv_out)
+
+
+def test_poltoind():
+    miriad_uv = UVData()
+    miriad_file = os.path.join(DATA_PATH, 'zen.2456865.60537.xy.uvcRREAA')
+    uvtest.checkWarnings(miriad_uv.read_miriad, [miriad_file], known_warning='miriad')
+    pol_arr = miriad_uv.polarization_array
+
+    miriad = miriad_uv._convert_to_filetype('miriad')
+    miriad.polarization_array = None
+    nt.assert_raises(ValueError, miriad._pol_to_ind, pol_arr[0])
+
+    miriad.polarization_array = [pol_arr[0], pol_arr[0]]
+    nt.assert_raises(ValueError, miriad._pol_to_ind, pol_arr[0])
```

Comment:
I think that's covered by the tests where actual files are read and processed, I added these tests to cover cases that weren't already covered. But I can add another case if you think I should. Or I can add a docstring or change the name of the test to make it clear I'm testing error handling.